src/Extracting/ParsesValidationRules.php

```diff
@@ -208,9 +209,13 @@ protected function parseRule($rule, array &$parameterData, bool $independentOnly
             if (enum_exists($type) && method_exists($type, 'tryFrom')) {
                 // $case->value only exists on BackedEnums, not UnitEnums
                 // method_exists($enum, 'tryFrom') implies $enum instanceof BackedEnum
-                // @phpstan-ignore-next-line
-                $cases = array_map(fn ($case) => $case->value, $type::cases());
-                $parameterData['type'] = gettype($cases[0]);
+                $cases = [];
+                foreach ($type::cases() as $case) {
+                    /** @var BackedEnum $case */
+                    $cases[$case->name] = $case->value;
+                }
+                
+                $parameterData['type'] = gettype(reset($cases));
```

Comment:
Hmm, what does this do? The `$cases` array is not used for anything else beyond getting the type, and it does the same thing as the previous version, which works for Backed Enums.